src/main/java/io/vertx/mqtt/impl/MqttEndpointImpl.java

```diff
@@ -438,11 +440,19 @@ public MqttEndpointImpl publish(String topic, Buffer payload, MqttQoS qosLevel,
 
   @Override
   public MqttEndpointImpl publish(String topic, Buffer payload, MqttQoS qosLevel, boolean isDup, boolean isRetain, Handler<AsyncResult<Integer>> publishSentHandler) {
+    return publish(topic, payload, qosLevel, isDup, isRetain, this.nextMessageId(), publishSentHandler);
+  }
+
+  @Override
+  public MqttEndpointImpl publish(String topic, Buffer payload, MqttQoS qosLevel, boolean isDup, boolean isRetain, int messageId, Handler<AsyncResult<Integer>> publishSentHandler) {
+    if(messageId > MAX_MESSAGE_ID || messageId < 0) {
```

Comment:
Just a couple of nits ...
Space after `if` and space before and after the `+` on string concatenation. Thanks!

Reply:
Sure, fixed that.